implementations/rust/ockam/ockam_api/src/nodes/models/portal.rs

```diff
@@ -237,10 +246,11 @@ impl OutletStatus {
 }
 
 /// Response body when returning a list of Inlets
-#[derive(Debug, Clone, Decode, Encode)]
+#[derive(Debug, Clone, Decode, Encode, Serialize)]
 #[rustfmt::skip]
 #[cbor(map)]
 pub struct InletList {
+	#[serde(rename = "inlets")]
     #[n(1)] pub list: Vec<InletStatus>
 }
```

Comment:
This still looks off to me... I think the proper way of doing this would be to delete all these `XList` structs that contain a single field with a vector, and refactor the endpoints that use them to return the plain vector. That way, you don't need to mess with serde anymore and you can customize the `ockam node show` structure as you want.

Since this is a little out of the scope for this PR, let me know if you want to tackle that as part of this PR. Otherwise I'll create a new issue for it.

Reply:
Yes, I completely agree. In fact, in my change to the inlets and outlets list I got rid of the list wrapper to reduce complexity and I think that it's way better. I agree that the change can be out of scope of the PR but I would gladly take on it after (or before) this one. You can create the issue and assign it to me.